src/core/request.js

```diff
@@ -36,18 +36,19 @@ import _ from 'lodash';
 import __ from 'i18n';
 import module from 'module';
 import context from 'context';
-import Promise from 'core/promise';
 import promiseQueue from 'core/promiseQueue';
 import tokenHandlerFactory from 'core/tokenHandler';
 import loggerFactory from 'core/logger';
 
-var tokenHeaderName = 'X-CSRF-Token';
+const tokenHeaderName = 'X-CSRF-Token';
 
-var tokenHandler = tokenHandlerFactory();
+const tokenHandler = tokenHandlerFactory();
 
-var queue = promiseQueue();
+const queue = promiseQueue();
 
-var logger = loggerFactory('core/request');
+const logger = loggerFactory('core/request');
+
+let tempToken;
```

Comment:
Caution, the variable is scoped to the whole module, which could lead to issues in a concurrent environment

Reply:
You are absolutely right - I have moved it to the narrowest scope it can have.